### PR TITLE
Allow Java 11 as LTS for older streams and other platforms

### LINKED ISSUE #38732: Quarkus should still allow to create project with Java 11 (for older streams and other platforms)
### Describe the bug

We removed 11 as LTS version, even if we new versions of Quarkus have 17 as min version, the tooling should still allow to create project with Java 11 using older stream or other platforms.

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #38732

### CODE REVIEW

**Comment by ia3andy** [comment]
@gsmet should I merge this?

**Comment by ia3andy** [comment]
CI passed I rebased (maybe I shouldn't have 😅)

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public String toString() {
     }
 
     // ordering is important here, so let's keep them ordered
-    public static final SortedSet<Integer> JAVA_VERSIONS_LTS = new TreeSet<>(List.of(17, 21));
+    public static final SortedSet<Integer> JAVA_VERSIONS_LTS = new TreeSet<>(List.of(11, 17, 21));
```
Comment (geoand) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java:65]: Wouldn't this mean that the Java 11 will be shown as an option when we release Quarkus 3.8 and it's an LTS, even though Java 17 will be the minimum version?

File: independent-projects/tools/devtools-common/src/test/java/io/quarkus/devtools/project/JavaVersionTest.java [hunk 2]
```diff
@@ -31,7 +31,8 @@ public void givenJavaVersion21ShouldReturn21() {
 
     @Test
     void shouldProperlyUseMinJavaVersion() {
-        assertThat(getCompatibleLTSVersions(new JavaVersion("17"))).isEqualTo(JAVA_VERSIONS_LTS);
+        assertThat(getCompatibleLTSVersions(new JavaVersion("11"))).isEqualTo(JAVA_VERSIONS_LTS);
+        assertThat(getCompatibleLTSVersions(new JavaVersion("17"))).containsExactly(17, 21);
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/test/java/io/quarkus/devtools/project/JavaVersionTest.java:35]: @geoand that should answer your question, this is what is used to get the list of Java versions, the minimum is set in the Platform/Stream.

So no 11 won't be in the list if the minimum Java version of the stream is 17.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java (hunk 1 above)
Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java:65]: @geoand did you see my comment?

Comment (geoand) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java:65]: I did

**Comment by ia3andy** [comment]
@gsmet this is also important for `3.7.3`

**Comment by ia3andy** [comment]
@geoand could you approve?

**Comment by geoand** [comment]
I don't know much of anything about this part of the code, so I'll abstain

**Review by gsmet — APPROVED** [review]

**Pull request merged**